### DISCUSSION TITLE
Is larq supported on x86 based CPUs?

### DISCUSSION THREAD

**Discussion by gj-raza** [larq/larq#673] [Q&A] [answered]
Can larq compute engine run models on x86 (say intel cpus) cpus or is it exclusive for ARM based cpus

**Comment by Tombana** [comment, marked as answer]
Hi @gj-raza,
Larq Compute Engine can run on any platform, including x86 cpus. However, *optimized* implementations are only provided for ARM Cortex-A systems (and ARM Cortex-M in a private fork). When you run Larq Compute Engine on x86, it will fall back to slower code and not use SIMD vector instructions, for example.

**Reply by jneeven** [reply]
(see also https://github.com/larq/larq/issues/607#issuecomment-753845989)

**Reply by gj-raza** [reply]
So does it mean that it wont also use AVX2/AVX512 SIMD instructions supported on x86 CPUs?

**Reply by Tombana** [reply]
Correct.

**Reply by gj-raza** [reply]
Got it. Thanks.

**Reply by Tombana** [reply]
To clarify: we don't have explicit assembly code or intrinsics to use such SIMD instructions on x86. It can still happen that the compiler that is very smart and uses them.
